antispam_bee.php

```diff
@@ -2299,6 +2333,21 @@ public static function add_spam_reason_to_comment( $comment_id )
 		);
 	}
 
+	public static function save_ip_hash( $comment_id, $approved, $comment_data )
+	{
+		$hashed_ip = self::hash_ip( $comment_data['comment_author_IP'] );
+		add_comment_meta(
+			$comment_id,
+			'antispam_bee_iphash',
+			$hashed_ip
+		);
+	}
+
+	public static function hash_ip($ip)
+	{
+		return substr( sha1( md5( self::$_salt . $ip ) ), 0, 10 );
```

Comment:
Hope it's clear that this is no real anonymization, just some added obscurity. Protection against a third party it is, but not for an admin or other plugins knowing the salt.

At least for IPv4 there is a high chance for no collision within the first 10 characters (depending on the salt) and the computation is lightweight, thus the real IP can be "calculated" within seconds on a modern consumer-grade computer.

Reply:
Thanks for your feedback @stklcode, you are right. I was thinking about changing this with  `wp_hash_password()` and using `wp_check_password()` somehow. But the current implementation shouldn't go into master.

Reply:
My current approach would be